pkg/util/ip_address.go

```diff
@@ -1,54 +1,77 @@
 package util
 
 import (
+	"fmt"
 	"net"
 	"strings"
+
+	"golang.org/x/xerrors"
 )
 
 // ParseIPAddress parses an IP address and removes port and/or IPV6 format
-func ParseIPAddress(input string) string {
-	host, _ := SplitHostPort(input)
+func ParseIPAddress(input string) (string, error) {
+	host, _, err := SplitHostPort(input)
+	if err != nil {
+		return "", xerrors.Errorf("Failed to split network address '%s' by host and port: %w",
+			input, err)
+	}
 
 	ip := net.ParseIP(host)
-
 	if ip == nil {
-		return host
+		return host, nil
 	}
 
 	if ip.IsLoopback() {
-		return "127.0.0.1"
+		if strings.Contains(host, ":") {
+			// IPv6
+			return "::1", nil
+		}
+		return "127.0.0.1", nil
 	}
 
-	return ip.String()
+	return ip.String(), nil
 }
 
 // SplitHostPortDefault splits ip address/hostname string by host and port. Defaults used if no match found
-func SplitHostPortDefault(input, defaultHost, defaultPort string) (host string, port string) {
-	port = defaultPort
-	s := input
-	lastIndex := strings.LastIndex(input, ":")
-
-	if lastIndex != -1 {
-		if lastIndex > 0 && input[lastIndex-1:lastIndex] != ":" {
-			s = input[:lastIndex]
-			port = input[lastIndex+1:]
-		} else if lastIndex == 0 {
-			s = defaultHost
-			port = input[lastIndex+1:]
+func SplitHostPortDefault(input, defaultHost, defaultPort string) (host string, port string, err error) {
```

Comment:
Instead of returning 3 result types maybe consider putting host and port into a struct and use that as result type? Reason is we don't normally have more than 2 return types. But I'm open to discuss it.

Reply:
@marefr OK I changed to using a struct `NetworkAddress` - please have a look.

Reply:
Looks great :+1: